pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -229,42 +227,36 @@ func podToEndpointAddressForService(svc *v1.Service, pod *v1.Pod) (*v1.EndpointA
 	var endpointIP string
 	ipFamily := v1.IPv4Protocol
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) {
-		// In a legacy cluster, the pod IP is guaranteed to be usable
-		endpointIP = pod.Status.PodIP
+	if len(svc.Spec.IPFamilies) > 0 {
+		// controller is connected to an api-server that correctly sets IPFamilies
+		ipFamily = svc.Spec.IPFamilies[0] // this works for headful and headless
 	} else {
-		//feature flag enabled and pods may have multiple IPs
-		if len(svc.Spec.IPFamilies) > 0 {
-			// controller is connected to an api-server that correctly sets IPFamilies
-			ipFamily = svc.Spec.IPFamilies[0] // this works for headful and headless
+		// controller is connected to an api server that does not correctly
+		// set IPFamilies (e.g. old api-server during an upgrade)
```

Comment:
should leave a fixme since this will be dead code in a few more releases?

Reply:
Yes. we should. I will leave this comment open as a reminder to leave a note.